ush/ufoeval/setup_phase3.py

```diff
@@ -0,0 +1,211 @@
+            self.f.write(f". {self.homegfs}/ush/load_fv3gfs_modules.sh\n")
+        else:
+            self.f.write(f". {self.homegfs}/ush/load_ufsda_modules.sh\n")
+        self.f.write("set -x\n")
+
+    def aprun(self):
+        """
+        Execute app
+        """
+
+        # cd to run directory
+        self.f.write("\n")
+        self.f.write(f"# cd to run directory\n")
+        self.f.write(f"cd {self.rundir}\n")
+
+        # copy or link app executable
+        self.f.write("\n")
+        if self.appcore == 'gsi':
+            self.f.write(f"# Copy executable\n")
+            self.f.write(f"cp -p {self.appexe} ./gsi.x\n")
+        else:
+            self.f.write(f"# Link executables\n")
+            self.f.write(f"ln -fs {self.appexe} ./gdas.x\n")
+            self.f.write(f"ln -fs {self.incexe} ./fv3jedi_fv3inc.x\n")
+
+        # execute app
+        self.f.write("\n")
+        self.f.write(f"# Execute app\n")
+        self.f.write(f"export OMP_NUM_THREADS={self.threads}\n")
+        self.f.write(f"ulimit -s unlimited\n")
+
+        if self.appcore == 'gsi':
+            aprun_command = f"srun -n {self.ntasks} --cpus-per-task={self.threads} ./gsi.x"
+        else:
+            aprun_command = f"srun -n {self.ntasks} --cpus-per-task={self.threads} ./gdas.x fv3jedi variational ./atmanlvar.yaml"
+
+        self.f.write(f"{aprun_command}\n")
+
+        # JEDI jobs convert cube sphere increments to gausssian grid for comparison
+        if self.appcore == 'jedi':
+            self.f.write("\n")
+            self.f.write(f"# Convert cube sphere increments to gaussian grid\n")
+            aprun_command = f"srun -n {self.ntasks} --cpus-per-task={self.threads} ./fv3jedi_fv3inc.x ./atmanlfv3inc.yaml"
+            self.f.write(f"{aprun_command}\n")
+
+    def close(self):
+        """
+        Flush and make the card executable
+        """
+        self.f.close()
+        subprocess.run(["chmod", "+x", jobname + ".sh"])
+
+
+def main():
+    epilog = ["Examples:",
+              "   ./run.py -c config.yaml",
+              "   ./run.py -h"]
+    parser = argparse.ArgumentParser(description="Set up atmospheric DA run.",
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     epilog=os.linesep.join(epilog))
+    parser.add_argument("-c", "--config", required=True, help="The YAML file")
+    args = parser.parse_args()
+
+    # Get the experiment configuration
+    run_jjobs_yaml = args.config
+    with open(run_jjobs_yaml, 'r') as file:
+        exp_config = yaml.safe_load(file)
+
+    logging.info(f"exp_config {exp_config}")
+
+    # Set DA core and type
+    appcore = exp_config['app files']['DA_CORE']
+    apptype = exp_config['app files']['DA_TYPE']
+
+    # Ensure valid app core and type
+    valid = ['gsi', 'jedi']
+    if appcore not in valid:
+        logging.error(f"DA_CORE {appcore} is invalid.  Valid cores are {valid}")
+        sys.exit()
```

Comment:
```suggestion
        raise ValueError(f"DA_CORE {appcore} is invalid.  Valid cores are {valid}")
```

Reply:
See: https://docs.python.org/3/library/exceptions.html#bltin-exceptions

Reply:
Thank you @CoryMartin-NOAA for the help.  I read through the link.  The list of exceptions is huge.  Change committed at 1a81235.

Reply:
Yes, but in practice there are usually only a few key ones used.